lib/Amazon/S3/Thin/Credentials.pm

```diff
@@ -37,6 +52,62 @@ sub new {
     return bless $self, $class;
 }
 
+sub from_env {
```

Comment:
Would you please add documentation about `from_env` and `from_metadata`? (what is version and role in a nutshell)